### DISCUSSION TITLE
`@keydown.space` listens for `_` and `-`

### DISCUSSION THREAD

**Discussion by marcreichel** [alpinejs/alpine#3124] [7. Bugs (archived)]
When I attach `@keydown.space.prevent` to an `input` element and now try to enter `_` or `-` into the input, nothing happens as the keystrokes get prevented.

This seems unintentional as I just want to prevent key presses of the space key.

Edit: Here's a Codepen demonstrating the issue: https://codepen.io/marcreichel/pen/VwxZGoJ

**Comment by ekwoka** [comment]
I've looked over the source and there is definitely a part of it that makes it block `-` on space, but I'll have to look into it deeper to see how it is then even blocking ` ` and `_`.

Some...interesting decisions made in the writing of the key handler code.

I'll submit a PR for it once I figure out a good solution.

EDIT: PR: https://github.com/alpinejs/alpine/pull/3134
